packages/nimbus/src/patterns/fields/date-range-picker-field/date-range-picker-field.stories.tsx

```diff
@@ -0,0 +1,244 @@
+    );
+
+    await step("Form field accessibility is properly configured", async () => {
+      const segments = helpers.getDateSegments();
+
+      // DateRangePicker group should exist and be accessible
+      const dateGroup = helpers.getDateRangeGroup();
+      await expect(dateGroup).toBeInTheDocument();
+
+      // All segments should be accessible
+      for (const segment of segments) {
+        await expect(segment).toHaveAttribute("role", "spinbutton");
+      }
+
+      // FormField should provide proper labeling via aria-labelledby or similar
+      // (The exact mechanism depends on how FormField connects labels to inputs)
+      const label = helpers.getLabel();
+      await expect(label).toBeInTheDocument();
+    });
+
+    await step("Date input functionality works", async () => {
+      const segments = helpers.getDateSegments();
+
+      // Type in first segment (start month)
+      await userEvent.click(segments[0]);
+      await userEvent.keyboard("12");
+
+      await waitFor(async () => {
+        await expect(segments[0]).toHaveAttribute("aria-valuenow", "12");
+      });
+
+      // Clear for next tests
+      await userEvent.keyboard("{Control>}a{/Control}");
+      await userEvent.keyboard("{Delete}");
+    });
+
+    await step("Calendar functionality is integrated", async () => {
+      await helpers.openCalendar();
+
+      // Calendar should be open
+      const calendar = within(document.body).getByRole("application");
+      await expect(calendar).toBeInTheDocument();
+
+      // Close calendar
+      await userEvent.keyboard("{Escape}");
+      await waitFor(async () => {
+        const calendarAfter = within(document.body).queryByRole("application");
+        await expect(calendarAfter).not.toBeInTheDocument();
+      });
+    });
+  },
+};
+
+export const WithErrors: Story = {
+  args: {
+    touched: true,
+    errors: { missing: true, format: true },
+    isRequired: true,
+  },
+  render: (args) => {
+    const [value, setValue] = useState<DateRange | null>(null);
+    return <DateRangePickerField {...args} value={value} onChange={setValue} />;
+  },
+  play: async ({ canvasElement, step }) => {
+    const canvas = within(canvasElement);
+    const helpers = createFieldHelpers(canvas);
+
+    await step(
+      "Error messages are displayed when field is touched",
+      async () => {
+        const errorAlert = canvas.queryByRole("alert");
+        await expect(errorAlert).toBeInTheDocument();
+
+        // Should show built-in error messages for known error types
+        const missingError = canvas.getByText(/field is required/i);
+        await expect(missingError).toBeInTheDocument();
+
+        const formatError = canvas.getByText(/Please enter a valid format/i);
+        await expect(formatError).toBeInTheDocument();
+      }
+    );
+
+    await step("Field structure remains intact with errors", async () => {
+      const label = helpers.getLabel();
+      await expect(label).toBeInTheDocument();
```

Comment:
`getAllByRole` will error if the label is not in the document, so no need for `toBeInTheDocument` here